server/tserver/src/main/java/org/apache/accumulo/tserver/log/DfsLogger.java

```diff
@@ -361,11 +361,16 @@ public static DFSLoggerInputStreams readHeaderAndReturnStream(FSDataInputStream
       AccumuloConfiguration conf) throws IOException {
     DataInputStream decryptingInput;
 
-    byte[] magic = DfsLogger.LOG_FILE_HEADER_V4.getBytes(UTF_8);
-    byte[] magicBuffer = new byte[magic.length];
+    byte[] magic4 = DfsLogger.LOG_FILE_HEADER_V4.getBytes(UTF_8);
+    byte[] magic3 = DfsLogger.LOG_FILE_HEADER_V3.getBytes(UTF_8);
+
+    if (magic4.length != magic3.length)
+      throw new IllegalStateException();
```

Comment:
This could use a message. Could also use AssertionError here, since it's being used that way.